### ISSUE TITLE
TypeError: fit() argument after ** must be a mapping, not NoneType

### ISSUE THREAD

**Issue by skepticleo** [databricks/spark-sklearn#63]
I am trying to run the sample code with iris dataset:

```
from sklearn import svm, datasets
from pyspark import SparkContext
from spark_sklearn import GridSearchCV
iris = datasets.load_iris()
parameters = {'kernel':('linear', 'rbf'), 'C':[1, 10]}
svr = svm.SVC()
sc = SparkContext()
clf = GridSearchCV(sc, svr, parameters)
clf.fit(iris.data, iris.target)
```

Running this throws a TypeError, 

```
Traceback (most recent call last):
"<stdin>", line 1, in <module>
"spark_sklearn/grid_search.py", line 272, in fit
return self._fit(X, y, groups, ParameterGrid(self.param_grid))
"spark_sklearn/grid_search.py", line 400, in _fit
best_estimator.fit(X, y, **fit_params)
TypeError: fit() argument after ** must be a mapping, not NoneType
```

```
Environment Info:
python 3.5.2
spark 2.2.0
spark_sklearn 0.2.0
```

**Comment by robegs** [comment]
I  have the same problem with 
python 2.7
spark 2.0
spark-sklearn
scikit-learn==0.19.0

**Comment by smurching** [comment]
Sorry about that, this issue due to spark-sklearn's use of scikit-learn APIs that were deprecated in scikit-learn 0.19.0 & will be resolved by #64.

For now a workaround is to manually set `clf.fit_params` to an empty dict if you don't intend to pass any extra params to the `fit` method of the estimator being cross-validated, e.g:

```
...
clf = GridSearchCV(sc, svr, parameters)
clf.fit_params = {}
clf.fit(iris.data, iris.target)
...
```
